src/threads/unfollow.ts

```diff
@@ -0,0 +1,31 @@
+import type { APIGatewayProxyHandler } from 'aws-lambda';
+import { dynamodb } from '../dynamodb';
+import { extractTokenData, getConnectionId } from '../utils/parsers';
+import { badRequest, notFound, ok, serverError, unauthorized } from '../utils/responses';
+import { ForumTable } from './table';
+
+const forumTable = new ForumTable(dynamodb);
+
+export const handler: APIGatewayProxyHandler = async event => {
+  const connectionId = getConnectionId(event);
+  if (!connectionId) return badRequest('connectionId is required');
+  const tokenData = extractTokenData(event);
+  if (!tokenData) return unauthorized();
+  const { participantId, itemId, userId } = tokenData;
+
+  try {
+    const [ followEvent ] = await forumTable.getFollowers({
+      participantId,
+      itemId,
+      limit: 1,
+      filters: { userId, connectionId },
+    });
+    if (!followEvent) return notFound();
+
+    await forumTable.removeThreadEvent(followEvent);
```

Comment:
There is no need to check anything. Just request the "follow" event removal. 
This should work (for your tests) either if the record exists or not, you'll never know if it exists, but that's not a problem. (anyway this is called when leaving the thread, the frontend does not care about the response)

Reply:
I need to know the sort key of the event to remove, that's why I search for it first.
I'll remove the not-found early return and update the tests accordingly

Reply:
Fixed

Reply:
I don't think you need the sort key to remove it, primary key and criterions should be enough. Doing two requests for such an operation is a bit against the way of working with such dbs.